lib/quicly.c

```diff
@@ -1854,14 +1839,25 @@ static int new_path(quicly_conn_t *conn, size_t path_index, struct sockaddr *rem
     return 0;
 }
 
-static void do_delete_path(quicly_conn_t *conn, struct st_quicly_conn_path_t *path)
+static int do_delete_path(quicly_conn_t *conn, struct st_quicly_conn_path_t *path)
 {
-    if (path->dcid != UINT64_MAX && conn->super.remote.cid_set.cids[0].cid.len != 0)
-        retire_connection_id(conn, path->dcid);
+    int ret;
+
+    if (path->dcid != UINT64_MAX && conn->super.remote.cid_set.cids[0].cid.len != 0) {
+        uint64_t cid = path->dcid;
+        dissociate_cid(conn, cid);
+        if ((ret = quicly_remote_cid_unregister(&conn->super.remote.cid_set, cid)) != 0)
+            return ret;
```

Comment:
What happens if this path is taken when `delete_path` is called from `quicly_free`? Don't we leak memory by skipping the invocation of `free(path);`?

Reply:
Addressed in cf2e0a5.